wgpu-hal/src/vulkan/mod.rs

```diff
@@ -1210,6 +1211,11 @@ impl crate::Queue for Queue {
             signal_values.push(!0);
         }
 
+        let mut pending_signal_semaphores = self.signal_semaphores.lock();
+        if !pending_signal_semaphores.is_empty() {
+            signal_semaphores.append(&mut pending_signal_semaphores);
```

Comment:
This also needs to push to signal_values or you'll throw a validation error as the values/semaphores won't be matched up.

Reply:
I am going to apply the comment.

Reply:
What if the caller wants to signal a timeline semaphore? Perhaps additional semaphore _values_ should be exposed as well?